File: src/Entity/Artifact.php
<?php

namespace App\Entity;

use App\Repository\ArtifactRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ArtifactRepository::class)]
class Artifact
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne]
    private Track $track;

    #[ORM\Column]
    private string $artifactName;

    #[ORM\Column(type: 'text')]
    private string $artifactDescription;

    #[ORM\Column]
    private string $artifactType;

    #[ORM\Column(nullable: true)]
    private ?string $language = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $testContent = null;

    #[ORM\Column]
    private \DateTimeImmutable $createdAt;

    //#[ORM\OneToMany(mappedBy: 'artifact', targetEntity: FileObject::class)]
    //private Collection $files;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(?int $id): void
    {
        $this->id = $id;
    }

    public function getTrack(): Track
    {
        return $this->track;
    }

    public function setTrack(Track $track): void
    {
        $this->track = $track;
    }

    public function getArtifactName(): string
    {
        return $this->artifactName;
    }

    public function setArtifactName(string $artifactName): void
    {
        $this->artifactName = $artifactName;
    }

    public function getArtifactDescription(): string
    {
        return $this->artifactDescription;
    }

    public function setArtifactDescription(string $artifactDescription): void
    {
        $this->artifactDescription = $artifactDescription;
    }

    public function getArtifactType(): string
    {
        return $this->artifactType;
    }

    public function setArtifactType(string $artifactType): void
    {
        $this->artifactType = $artifactType;
    }

    public function getLanguage(): ?string
    {
        return $this->language;
    }

    public function setLanguage(?string $language): void
    {
        $this->language = $language;
    }

    public function getTestContent(): ?string
    {
        return $this->testContent;
    }

    public function setTestContent(?string $testContent): void
    {
        $this->testContent = $testContent;
    }

    public function getCreatedAt(): \DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeImmutable $createdAt): void
    {
        $this->createdAt = $createdAt;
    }

    public function getFiles(): Collection
    {
        return $this->files;
    }

    public function setFiles(Collection $files): void
    {
        $this->files = $files;
    }



}

File: src/entities/Artifact.java
package entities;

import java.time.LocalDateTime;

public class Artifact {
    private int id;
    private int trackId;

    private String artifactName;
    private String artifactDescription;
    private String artifactType; // CODE/IMAGE/VIDEO/DOCUMENT/LINK/TEXT
    private String language;     // optional for CODE

    // In your SQL it’s test_content (typo), keep Java name clean:
    private String textContent;

    private LocalDateTime createdAt;
    private LocalDateTime deletedAt;

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getTrackId() { return trackId; }
    public void setTrackId(int trackId) { this.trackId = trackId; }

    public String getArtifactName() { return artifactName; }
    public void setArtifactName(String artifactName) { this.artifactName = artifactName; }

    public String getArtifactDescription() { return artifactDescription; }
    public void setArtifactDescription(String artifactDescription) { this.artifactDescription = artifactDescription; }

    public String getArtifactType() { return artifactType; }
    public void setArtifactType(String artifactType) { this.artifactType = artifactType; }

    public String getLanguage() { return language; }
    public void setLanguage(String language) { this.language = language; }

    public String getTextContent() { return textContent; }
    public void setTextContent(String textContent) { this.textContent = textContent; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getDeletedAt() { return deletedAt; }
    public void setDeletedAt(LocalDateTime deletedAt) { this.deletedAt = deletedAt; }

    public boolean isDeleted() { return deletedAt != null; }
}
